pkg/cli/admin/oauth/oauth.go

```diff
@@ -0,0 +1,261 @@
+	cmd.Flags().BoolVar(&o.RoleAll, "role-all", o.RoleAll, "Sets the role scope over all namespaces.")
+	cmd.Flags().StringSliceVar(&o.UserScopes, "scopes", o.UserScopes, "The scopes to be assigned to the user.")
+	cmd.Flags().DurationVar(&o.Duration, "duration", o.Duration, "How long before this token expires.")
+
+	kcmdutil.AddDryRunFlag(cmd)
+	o.PrintFlags.AddFlags(cmd)
+	return cmd
+}
+
+func (o *CreateTokenOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	var err error
+	o.DryRunStrategy, err = kcmdutil.GetDryRunStrategy(cmd)
+	if err != nil {
+		return err
+	}
+	o.Output = kcmdutil.GetFlagString(cmd, "output")
+
+	o.ToPrinter = func(message string) (printers.ResourcePrinter, error) {
+		o.PrintFlags.NamePrintFlags.Operation = message
+		kcmdutil.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.DryRunStrategy)
+		return o.PrintFlags.ToPrinter()
+	}
+
+	clientConfig, err := f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	userClient, err := userclient.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
+	oauthClient, err := oauthclient.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
+	o.UserInterface = userClient.UserV1().Users()
+	o.OAuthAccessTokenInterface = oauthClient.OauthV1().OAuthAccessTokens()
+	o.OAuthClientInterface = oauthClient.OauthV1().OAuthClients()
+
+	if len(o.UserUID) == 0 {
+		userName := o.UserName
+		if len(userName) == 0 {
+			userName = "~"
+		}
+		user, err := o.UserInterface.Get(context.Background(), userName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if len(user.UID) == 0 {
+			return fmt.Errorf("user %s must not be a virtual user (e.g. system:admin) - create a user first", user.Name)
+		}
+		o.UserName = user.Name
+		o.UserUID = string(user.UID)
+	}
+
+	if len(o.RoleScope) > 0 {
+		if o.RoleAll {
+			o.UserScopes = append(o.UserScopes, fmt.Sprintf("role:%s:*", o.RoleScope))
+		} else {
+			ns, _, err := f.ToRawKubeConfigLoader().Namespace()
+			if err != nil {
+				return err
+			}
+			o.UserScopes = append(o.UserScopes, fmt.Sprintf("role:%s:%s", o.RoleScope, ns))
+		}
+	}
+	if len(o.UserScopes) == 0 {
+		o.UserScopes = append(o.UserScopes, "user:full")
+	}
+
+	// use of this redirectURI prohibits normal login
+	o.RedirectURI = "https://localhost:8443/oauth/token/implicit"
+
+	return nil
+}
+
+func (o *CreateTokenOptions) Run() error {
+	if o.DryRunStrategy != kcmdutil.DryRunClient {
+		client := &oauthv1.OAuthClient{
+			ObjectMeta:  metav1.ObjectMeta{Name: o.OAuthClientName},
+			GrantMethod: oauthv1.GrantHandlerAuto,
+		}
+		if _, err := o.OAuthClientInterface.Create(context.Background(), client, metav1.CreateOptions{}); err != nil {
+			if !errors.IsAlreadyExists(err) {
+				return err
+			}
+		}
+	}
+
+	tokenBytes := make([]byte, 64)
+	if _, err := io.ReadAtLeast(rand.Reader, tokenBytes, 64); err != nil {
+		return err
+	}
+	accessToken := base64.RawURLEncoding.EncodeToString(tokenBytes)
```

Comment:
The code that generates tokens for the oauth-server prevents leading dashes, that might be something to consider if the main use of this is for scripts: https://github.com/openshift/oauth-server/blob/14ca5c7dcdfbbc4bb7935de071247ccc4f8499bf/pkg/osinserver/tokengen.go#L36-L36